java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture2/tasks/_2/Number2.java

```diff
@@ -0,0 +1,229 @@
+        scanner.nextInt();
+        System.out.println();
+        System.out.println("Please input a coordinates for robot.\nThere must be smaller than your room :)");
+        System.out.println("Input X and then Y");
+        robotGame.setCoordinates(scanner.nextInt(), scanner.nextInt());
+        System.out.println("Your coordinates are:");
+        robotGame.getCoordinates();
+        System.out.println();
+        System.out.println("Press 0 to continue");
+        System.out.println();
+        System.out.println("Set direction where your robot is going");
+        System.out.println("0. SOUTH,\n1. NORTH,\n2. WEST,\n3. EAST");
+        Direction direction = Direction.values()[Integer.parseInt(scanner.next())];
+
+        switch (direction.toString()) {
+            case "SOUTH" -> {
+                System.out.println("Direction is SOUTH\n");
+                robotGame.setDirection(Direction.SOUTH);
+            }
+            case "NORTH" -> {
+                System.out.println("Direction is NORTH\n");
+                robotGame.setDirection(Direction.NORTH);
+            }
+            case "WEST" -> {
+                System.out.println("Direction is WEST\n");
+                robotGame.setDirection(Direction.WEST);
+            }
+            case "EAST" -> {
+                System.out.println("Direction is EAST\n");
+                robotGame.setDirection(Direction.EAST);
+            }
+            default -> throw new InputMismatchException("Wrong input");
+        }
+
+        System.out.println("1.Move Forward.\n2.ChangeDirection;\n3.Get coordinates;\n4.Get directions;\n5.Exit");
+
+        while (!exit) {
+            switch (scanner.nextInt()) {
+                case 1:
+                    robotGame.moveForward();
+                    System.out.println("Moving On");
+                    break;
+                case 2:
+                    System.out.println("1.Turn Left;\n2.Turn Right;");
+                    if (scanner.nextInt() == 1) {
+                        robotGame.turnLeft();
+                        System.out.println("Turned left");
+
+                    }
+                    else  {
+                        robotGame.turnRight();
+                        System.out.println("Turned right");
+                    }
+                    break;
+                case 3:
+                    robotGame.getCoordinates();
+                    break;
+                case 4:
+                    robotGame.getDirection();
+                    break;
+                case 5:
+                    System.out.println("Shutting down");
+                    exit = true;
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+}
+
+enum Direction {
+    SOUTH,
+    NORTH,
+    WEST,
+    EAST
+}
+
+class RobotGame {
+
+    private int w;
+    private int h;
+    private int x;
+    private int y;
+    private int[] coordinates = {x, y};
```

Comment:
`x` and `y` are actually defined twice - as separate fields and as an array. Could you please remove one declaration?